WalletWasabi/Blockchain/TransactionProcessing/TransactionProcessor.cs

```diff
@@ -61,6 +65,27 @@ public IEnumerable<ProcessedResult> Process(IEnumerable<SmartTransaction> txs)
 		return rets;
 	}
 
+	/// <summary>
+	/// Gets the wallet transaction with the given txid, if the transaction exists.
+	/// </summary>
+	public bool TryGetTransaction(uint256 txid, [NotNullWhen(true)] out SmartTransaction? smartTransaction)
```

Comment:
I don't see a reason to place this method here. It has nothing to do with the responsibility of processing transactions. It looks more like a wallet thing.